### ISSUE TITLE
pip-tools fails to install in virtualenv with --system-site-packages and pip==8.0.0

### ISSUE THREAD

**Issue by chaoflow** [nvie/pip-tools#302]
**old title**: uninstall fail with local pip-sync with pip=8.0.0 in activated virtualenv with system-site-packages
Given a system-wide installed docutils exists:
```
% virtualenv -p python2.7 --system-site-packages .
% source bin/activate
% pip install 'pip-sync=1.4.4'
% pip install -U 'pip==8.0.0' || true     # uninstalling pip-7.1.2 fails to delete temp files, unrelated issue
% echo nose > req.txt
% pip-sync req.txt
...
```
pip-sync attempts to remove the system-wide docutils.

**Comment by chaoflow** [comment]
new title: pip fails to install in virtualenv with --system-site-packages and pip==8.0.0
```
% virtualenv -p python2.7 --system-site-packages .
% source bin/activate
% pip install --upgrade 'pip==8.0.0'
% pip install --upgrade 'pip-tools==1.4.4'
Collecting pip-tools==1.4.4
  Downloading pip_tools-1.4.4-py2.py3-none-any.whl
Collecting six (from pip-tools==1.4.4)
  Downloading six-1.10.0-py2.py3-none-any.whl
Collecting click>=6 (from pip-tools==1.4.4)
  Downloading click-6.2-py2.py3-none-any.whl (70kB)
Collecting first (from pip-tools==1.4.4)
  Downloading first-2.0.1-py2.py3-none-any.whl
Installing collected packages: six, click, first, pip-tools
  Found existing installation: six 1.5.2
Detected a distutils installed project ('six') which we cannot uninstall. The metadata provided by distutils does not contain a list of files which have been installed, so pip does not know which files to uninstall.
```

**Comment by chaoflow** [comment]
Without --system-site-packages it installs fine.

**Comment by nvie** [comment]
Yeah, the packaging tools sport a lot of flags and options, making for many edge cases. I've never heard about `--system-site-packages` until now. Can those packages be detected from the output of `pip list`? If not, this is going to be tough to support. (I'm also not aiming to support _all_ use cases, only the ones that will not turn the pip-tools source into a big hairball.)

(Btw, I just released pip==1.5 with pip8 support, but this problem is unaffected by it.)

**Comment by chaoflow** [comment]
With pip-8.0.2 and pip-tools-1.6 this error is gone
